apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,132 @@
+  const user = await selectUserByUid(uid);
+  if (!user) return false;
+
+  const enrollment = await getEnrollmentForUserInCourseInstance({
+    user_id: user.user_id,
+    course_instance_id: course_instance_id,
+  });
+  if (!enrollment) return null;
+  return user;
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+    });
+    res.locals.policies = result.rows;
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
+    if (req.body.__action === 'add_new_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+      const params = {
+        assessment_id: res.locals.assessment.id,
+        created_by: res.locals.authn_user.user_id,
+        credit: req.body.credit,
+        end_date: new Date(req.body.end_date),
+        group_name: req.body.group_name || null,
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        group_id: group_id || null,
+        user_id: user_id || null,
+      };
+
+      await sqldb.queryAsync(sql.insert_assessment_access_policy, params);
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'delete_override') {
+      await sqldb.queryAsync(sql.delete_assessment_access_policy, {
+        assessment_id: res.locals.assessment.id,
+        unsafe_assessment_access_policies_id: req.body.policy_id,
+      });
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'edit_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+
+      const edit_params = {
+        assessment_id: res.locals.assessment.id,
+        credit: req.body.credit,
+        end_date: new Date(req.body.end_date),
+        group_name: req.body.group_name || null,
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        group_id: group_id || null,
+        user_id: user_id || null,
+        assessment_access_policies_id: req.body.policy_id,
+      };
+
+      await sqldb.queryAsync(sql.update_assessment_access_policy, edit_params);
```

Comment:
Conventionally, we don't bother declaring params separately if they're only used in one place.


```suggestion
      await sqldb.queryAsync(sql.update_assessment_access_policy, {
        assessment_id: res.locals.assessment.id,
        credit: req.body.credit,
        end_date: new Date(req.body.end_date),
        group_name: req.body.group_name || null,
        note: req.body.note || null,
        start_date: new Date(req.body.start_date),
        group_id: group_id || null,
        user_id: user_id || null,
        assessment_access_policies_id: req.body.policy_id,
      });
```

Reply:
Fixed in c5e27c6d4